homeassistant/components/ecobee/climate.py

```diff
@@ -564,14 +595,29 @@ def hvac_action(self):
     def extra_state_attributes(self) -> dict[str, Any] | None:
         """Return device specific state attributes."""
         status = self.thermostat["equipmentStatus"]
-        return {
+        if self.has_aux_heat:
+             return {
             "fan": self.fan,
             "climate_mode": self.comfort_settings.get(
                 self.thermostat["program"]["currentClimateRef"]
             ),
             "equipment_running": status,
             "fan_min_on_time": self.settings["fanMinOnTime"],
+            "aux_cutover_threshold": TemperatureConverter.convert(
+                self.settings["compressorProtectionMinTemp"] / 10,
+                UnitOfTemperature.FAHRENHEIT,
+                self.hass.config.units.temperature_unit,
+            )
```

Comment:
If we know the current value, why don't we make this a number entity instead? That way, we do not need a service and we have an actual entity to work with.

Reply:
Why not a number entity? Because I copied the code from the fan runtime. (SERVICE_SET_FAN_MIN_ON_TIME). I think this is because of the upstream library? I’m very much learning how this all works (python and HASS) so my apologies if it’s obvious. 

I don’t have the ability to test this code for a little while.

Reply:
> I think this is because of the upstream library?

The code shows the upstream library has the features.

> I don’t have the ability to test this code for a little while.

Well this is not about testing, but about implementing a number platform for this integration.

We, unfortunately, cannot accept the PR in the with the current implementation, simply because we are moving away from state attributes for cases like this.

Would you be able to implement that?

../Frenck

Reply:
I’ll give it a shot. I suspect that’s a whole new PR what with updating documentation. 

The principle is that the aux heat threshold shows as a wholly different entity as part of the overall ecobee integration, instead of a property on the climate entity, correct? Similar to `sensor.home_assistant_host_os_agent_version` in the Supervisor integration for example.

Reply:
Yes, except the example you gave is the `sensor` platform, in this case I'd expect the `number` platform to be used (as sensors are read-only, while this aux heat threadhold can be set as well).

Reply:
Number Entities are the way to model read/write numerical data.  This is the same feedback provided in April.

https://github.com/home-assistant/core/pull/114781#issuecomment-2041280235